hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/CallFeature.java

```diff
@@ -312,12 +396,321 @@ public void ethCallReentrancyCallFunction() {
+        assertThat(statusAfterAssociate).isEqualTo(22);
+        // transfer after dissociate should fail > response code 184 equals to owner does not own the token
+        assertThat(statusAfterDissociate).isEqualTo(184);
+    }
+
+    @Then("I dissociate a NFT and fail transfer")
+    public void ethCallAssociateNftTokenDissociateFailTransfer() {
+        var data = encodeData(
+                PRECOMPILE,
+                DISSOCIATE_TOKEN_FAIL_TRANSFER,
+                asAddress(nonFungibleTokenId),
+                asAddress(receiverAccountId),
+                asAddress(secondReceiverAlias),
+                new BigInteger("0"),
+                new BigInteger("1"));
+        var response = callContract(data, precompileContractAddress);
+        var statusAfterAssociate = response.getResultAsListDecimal().get(0);
+        var statusAfterDissociate = response.getResultAsListDecimal().get(1);
+
+        // transfer after associate should pass -> response code 22 equals SUCCESS
+        assertThat(statusAfterAssociate).isEqualTo(22);
+        // transfer after dissociate should fail -> response code 237 equals to owner does not own the NFT
+        assertThat(statusAfterDissociate).isEqualTo(184);
+    }
+
+    @Then("I approve a FUNGIBLE token and transfer it")
+    public void ethCallApproveFungibleTokenAndTransfer() {
+        networkTransactionResponse = tokenClient.associate(thirdReceiver, fungibleTokenId);
+        verifyMirrorTransactionsResponse(mirrorClient, 200);
+
+        var data = encodeData(
+                PRECOMPILE,
+                APPROVE_FUNGIBLE_TOKEN_AND_TRANSFER,
+                asAddress(fungibleTokenId),
+                asAddress(thirdReceiver),
+                new BigInteger("1"));
+        var response = callContract(data, precompileContractAddress);
+        var results = response.getResultAsListDecimal();
+        assertThat(results).isNotNull().hasSize(4);
+
+        // allowance before transfer should equal the amount
+        assertThat(intValue(results.get(0))).isZero();
+        // balance before + amount should equal the balance after
+        assertThat(intValue(results.get(1)) + 1).isEqualTo(intValue(results.get(3)));
+        // allowance after transfer should be 0
+        assertThat(intValue(results.get(2))).isZero();
+    }
+
+    @Then("I approve a NFT token and transfer it")
+    public void ethCallApproveNftTokenAndTransfer() {
+        var data = encodeData(
+                PRECOMPILE,
+                APPROVE_NFT_TOKEN_AND_TRANSFER,
+                asAddress(nonFungibleTokenId),
+                asAddress(receiverAccountId),
+                new BigInteger("1"));
+        var response = callContract(data, precompileContractAddress);
+        var results = response.getResultAsListAddress();
+
+        assertThat(results).isNotNull().hasSize(4);
+
+        // allowed address before transfer should be the receiverAccount
+        assertThat(results.get(0)).isEqualTo(results.get(3));
```

Comment:
nit: Comments like these would be more useful as assertj descriptions so they show up in test failure messages.
```suggestion
        assertThat(results.get(0)).as("allowed address before transfer should be the receiverAccount").isEqualTo(results.get(3));
```

Reply:
Great idea! Updated all tests with these kind of comments as suggested and it looks way better